source/connection-monitoring-and-pooling/tests/pool-checkout-maxConnecting-is-enforced.yml

```diff
@@ -18,29 +18,30 @@ poolOptions:
   maxPoolSize: 10
   waitQueueTimeoutMS: 5000
 operations:
-  # start creating a Connection. This will take a while
-  # due to the fail point.
+  # start 3 threads
   - name: start
     target: thread1
+  - name: start
+    target: thread2
+  - name: start
+    target: thread3
+  # start creating a Connection. This will take a while
+  # due to the fail point.
   - name: checkOut
     thread: thread1
-  # Start 2 new threads that wait for a little then try
-  # to check out connections. Only one thread should
+  # wait for thread1 to actually start creating a Connection
+  - name: waitForEvent
+    event: ConnectionCreated
+    count: 1
+  # wait some more time to ensure thread1's Connection is ready first
```

Comment:
The waiting here is to ensure thread1 has begun establishing a connection, not that the connection is ready (aka established).

Reply:
Fixed.